internal/overlord/servstate/handlers.go

```diff
@@ -601,6 +612,31 @@ func (s *serviceData) stop() error {
 	return nil
 }
 
+func (s *serviceData) waitProcessGroup() error {
```

Comment:
Two small things:

1) I'd be inclined to call this `waitChildren()` now as it's waiting on all children, not just the process group -- let's make the name reflect what is does clearly
2) Let's make it a standalone function `func waitChildren(pid int) error { ... }`, as it only depends on `s.cmd.Process.Pid` from `s`, so it's cleaner to pass just the dependencies we use.

Reply:
Agree on both points, good remarks! Adjusted.